apps/web/src/services/asset-inventory/collector/collector-history/CollectorHistoryPage.vue

```diff
@@ -121,259 +120,180 @@ import ErrorHandler from '@/common/composables/error/errorHandler';
 import { useManagePermissionState } from '@/common/composables/page-manage-permission';
 import { useQueryTags } from '@/common/composables/query-tags';
 
-import { peacock, green, red } from '@/styles/colors';
-
+import {
+    statusIconColorFormatter,
+    statusIconFormatter,
+    statusTextColorFormatter,
+    statusTextFormatter,
+} from '@/services/asset-inventory/collector/collector-history/lib/formatter-helper';
 import NoCollectorModal from '@/services/asset-inventory/collector/collector-history/modules/NoCollectorModal.vue';
 import { JOB_SELECTED_STATUS } from '@/services/asset-inventory/collector/collector-history/type';
 import { JOB_STATE } from '@/services/asset-inventory/collector/type';
 import { ASSET_INVENTORY_ROUTE } from '@/services/asset-inventory/route-config';
 
-const PROGRESS_BAR_COLOR = peacock[500];
-const COMPLETED_ICON_COLOR = green[500];
-const FAILED_ICON_COLOR = red[400];
-
-const statusTextFormatter = (status) => {
-    if (status === JOB_STATE.SUCCESS) return i18n.t('INVENTORY.COLLECTOR.HISTORY.SUCCESS');
-    if (status === JOB_STATE.IN_PROGRESS) return i18n.t('MANAGEMENT.COLLECTOR_HISTORY.MAIN.IN_PROGRESS');
-    if (status === JOB_STATE.CANCELED) return i18n.t('INVENTORY.COLLECTOR.HISTORY.CANCELED');
-    if (status === JOB_STATE.FAILURE) return i18n.t('INVENTORY.COLLECTOR.HISTORY.FAILURE');
-    return capitalize(status);
-};
-const statusTextColorFormatter = (status) => {
-    if ([JOB_STATE.CANCELED, JOB_STATE.FAILURE].includes(status)) return FAILED_ICON_COLOR;
-    return undefined;
+const valueHandlerMap: ValueHandlerMap = {
+    job_id: makeDistinctValueHandler('inventory.Job', 'job_id'),
+    status: makeEnumValueHandler(JOB_STATE),
+    collector_id: makeReferenceValueHandler('inventory.Collector'),
 };
```

Comment:
suggestion: Although it's not a responsive object, having `valueHandlerMap` under `handlers` might make it easier to understand.